pkg/broker/constants.go

```diff
@@ -1,18 +1,58 @@
 package broker
 
-import "time"
+import (
+	"time"
+)
 
 // CacheTTL TTL for catalog cache record expiry
 var CacheTTL = 1 * time.Hour
 
 var nonCfnParams = []string{
-	"aws_access_key",
-	"aws_secret_key",
 	"region",
 	"target_role_name",
 	"target_account_id",
 }
 
+var nonCfnParamDefs = map[string]interface{}{
+	"target_account_id": map[string]interface{}{
+		"description":   "AWS Account ID to provision into",
+		"display_group": "AWS Account Information",
+		"title":         "AWS Target Account ID",
+		"type":          "string",
+		"default":       "",
+	},
+	"target_role_name": map[string]interface{}{
+		"description":   "AWS IAM Role name to use for provisioning",
+		"display_group": "AWS Account Information",
+		"title":         "AWS Target Role Name",
+		"type":          "string",
+		"default":       "",
+	},
+	"region": map[string]interface{}{
+		"description":   "AWS Region to provision into",
+		"display_group": "AWS Account Information",
+		"title":         "AWS Region",
+		"type":          "string",
+		"default":       "us-west-2",
```

Comment:
What do you think about removing the default and making `region` a required param?

Reply:
Agreed, I think I put this in to try and minimize backward incompatible changes, but considering there are other changes that break backwards incompatible changes, might as well put it in now.

Reply:
with the default removed provisioning without specifying a region param results in stack being deployed to the region specified by the `-region` cli argument. @vsomayaji What are your thoughts on this? think we should rather be failing with a missing parameter error ?